### ISSUE TITLE
no visible global variables (object-usage-linter) is still a matter

### ISSUE THREAD

**Issue by milanglacier** [r-lib/lintr#1127]
`packageVersion('lintr')`: `‘2.0.1.9000’`
I installed `lintr` via github and I am at commit `4ca7e484`

The following example will yield the object-usage-linter error:

```R
library(tidyverse)

dog = function() {
    a = rnorm(5) %>% sum()
    a
}

```

yields `no visible global function definition for ‘%>%’ object_usage_linter [4, 1]`

`R --version`

```
R version 4.1.2 (2021-11-01) -- "Bird Hippie"
Copyright (C) 2021 The R Foundation for Statistical Computing
Platform: aarch64-apple-darwin20 (64-bit)
```

**Comment by AshesITR** [comment]
Thanks for reporting this. You can circumvent the issue by loading all packages before linting, but if the library call is in the same file as the function, we can do something about it.

**Comment by milanglacier** [comment]
> Thanks for reporting this. You can circumvent the issue by loading all packages before linting, but if the library call is in the same file as the function, we can do something about it.

Yes, the library call is definitely in the same file as the function.
I don't understand what is ` circumvent the issue by loading all packages before linting`,
does it mean that I fire the R process that runs the language server with those packages loaded?
like `R --slave -e "source('loadingPackage.R'); rlanguageserver::run()"`?

**Comment by AshesITR** [comment]
Yes, that should work.

**Comment by AshesITR** [comment]
Implementation should be straight-forward once #239 is merged.
That PR provides the necessary functions to find all exported objects, which could then be passed to `checkUsage()` as known globals.
